test-infra/src/main/java/com/datastax/oss/driver/api/testinfra/ccm/CcmBridge.java

```diff
@@ -139,18 +155,45 @@ private CcmBridge(
       allJvmArgs.append(quote);
     }
     this.jvmArgs = allJvmArgs.toString();
+    this.dseWorkloads = dseWorkloads;
+  }
+
+  public Optional<CassandraVersion> getDseVersion() {
+    return dse ? Optional.of(version) : Optional.empty();
   }
 
   public CassandraVersion getCassandraVersion() {
-    return cassandraVersion;
+    if (!dse) {
+      return version;
+    } else {
+      CassandraVersion cassandraVersion = version;
+      if (cassandraVersion.getMajor() == 6) {
+        // 6.0 = 4.0.x
+        return CassandraVersion.parse("4.0.0");
+      } else if (cassandraVersion.getMajor() == 5) {
+        if (cassandraVersion.getMinor() == 0) {
+          // 5.0 = 3.0.x
+          return CassandraVersion.parse("3.0.15");
+        } else {
+          // 5.1 == 3.10
+          return CassandraVersion.parse("3.10");
+        }
+      } else {
+        // assume anything older is 2.1.19
+        return CassandraVersion.parse("2.1.19");
+      }
```

Comment:
No big deal but I think it would be slightly more readable using constants and comparisons:

```java
CassandraVersion stableVersion = version.nextStable();
if (stableVersion.compareTo(V6_0_0) >= 0) {
  return CassandraVersion.parse("4.0.0");
} else if (stableVersion.compareTo(V5_1_0) >= 0) {
  return CassandraVersion.parse("3.10");
} else if (stableVersion.compareTo(V5_0_0) >= 0) {
  return CassandraVersion.parse("3.0.15");
} else {
  return CassandraVersion.parse("2.1.19");
}
```
And then the comments are not needed.